src/Workspaces/Remote/ServiceHub/Host/RemoteExportProviderBuilder.cs

```diff
@@ -93,15 +93,20 @@ public Assembly LoadAssembly(AssemblyName assemblyName)
         public Assembly LoadAssembly(string assemblyFullName, string? codeBasePath)
         {
             var assemblyName = new AssemblyName(assemblyFullName);
-            if (!string.IsNullOrEmpty(codeBasePath))
+            try
             {
-                // Set the codebase path, if known, as a hint for the assembly loader.
-#pragma warning disable SYSLIB0044 // https://github.com/dotnet/roslyn/issues/71510
-                assemblyName.CodeBase = codeBasePath;
-#pragma warning restore SYSLIB0044
+                // Attempt to load the assembly by its name.
+                return LoadAssembly(assemblyName);
             }
-
-            return LoadAssembly(assemblyName);
+#if NET
+            catch when (codeBasePath is not null)
+            {
+                // If that fails, try and load it from the codeBasePath if present
+                var selfAlc = System.Runtime.Loader.AssemblyLoadContext.GetLoadContext(Assembly.GetExecutingAssembly())!;
+                return selfAlc.LoadFromAssemblyPath(codeBasePath);
```

Comment:
We'd have to think about what this means for dependnecies though of your MEF component. Should we put you generally into an ALC? We do similar things in VS Code.